sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/IPartitionProcessorFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Creates instances of <see cref="IPartitionProcessor" />.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   An instance of a class implementing this interface is provided by the user in the <see cref="EventProcessor" />
+    ///   constructor.
+    /// </remarks>
+    ///
+    public interface IPartitionProcessorFactory
```

Comment:
I think an interface is overkill here;  If we're not comfortable using a `Func` in the constructor, I'd recommend making this a delegate instead.   That would allow more flexibility for callers while maintaining the same level of clarity and type safety.